default/python/AI/Character/Character.py

```diff
@@ -567,15 +567,17 @@ def func(self, *args, **kwargs):
 
 # Create combiners for traits that averages all not None results
 def average_not_none(llin):
-    ll = [x for x in llin if x]
+    ll = [x for x in llin if x != None]
+    if not ll:
+        return 0
     return sum(ll) / len(ll)
 
 for funcname in ["attitude_to_empire"]:
     setattr(Character, funcname, _make_single_function_combiner(funcname, average_not_none))
 
 # Create combiners for traits that use the geometic mean of all not None results
 def geometric_mean_not_none(llin):
-    ll = [x for x in llin if x]
+    ll = [x for x in llin if x != None]
     if not ll:
         return 1
     return math.exp( sum(map(math.log, ll)) / len(ll))
```

Comment:
Please add a sanity check if input values are greater than zero. Filter and log an error if not.